Add "Integration" pages to the getting started guide

dannytsang opened the pull request:
## Proposed change
Added a new *Concepts and terminology* page to highlight key things in Home Assistant. This will hopefully set the base line understanding for what are **devices and entities** for example.

I have also added an integration page with more detailed steps to adding an integration as an example.

## Type of change
<!--
    What types of changes does your PR introduce to our documentation/website?
    Put an `x` in the boxes that apply. You can also fill these out after
    creating the PR.
-->

- [x] Spelling, grammar or other readability improvements (`current` branch).
- [ ] Adjusted missing or incorrect information in the current documentation (`current` branch).
- [ ] Added documentation for a new integration I'm adding to Home Assistant (`next` branch).
  - [ ] I've opened up a PR to add logo's and icons in [Brands repository](https://github.com/home-assistant/brands).
- [ ] Added documentation for a new feature I'm adding to Home Assistant (`next` branch).
- [ ] Removed stale or deprecated documentation.

## Additional information
<!--
    Details are important, and help maintainers processing your PR.
    Please be sure to fill out additional details, if applicable.
-->

- Link to parent pull request in the codebase: 
- Link to parent pull request in the Brands repository: 
- This PR fixes or closes issue: fixes #22952

## Checklist
<!--
    Put an `x` in the boxes that apply. You can also fill these out after
    creating the PR. If you're unsure about any of them, don't hesitate to ask.
    We're here to help! This is simply a reminder of what we are going to look
    for before merging your code.
-->

- [x] This PR uses the correct branch, based on one of the following:
  - I made a change to the existing documentation and used the `current` branch.
  - I made a change that is related to an upcoming version of Home Assistant and used the `next` branch.
- [x] The documentation follows the Home Assistant documentation [standards].

[standards]: https://developers.home-assistant.io/docs/documenting/standards

dannytsang committed:
Added overview and integration page.

The overview outlines general concepts in Home Assistant such as what an integration is.

The Integration page contains steps that was at the end of the onboarding page. More detailed break to add an integration needs to be added.

dannytsang committed:
Added new images for device list and add-ons.

dannytsang committed:
Replaced device list with home assistant device page.

dannytsang committed:
Replaced overview with concepts and terminology.

Added screenshots to the page.

dannytsang committed:
Added images to adding Google Cast.

dannytsang committed:
Updated integrations with Google Cast as an example.

dannytsang committed:
Merge branch 'home-assistant:current' into 22952

dannytsang committed:
Refined some of wording.

dannytsang committed:
Fixed Google Cast punctuation.

dannytsang committed:
Merge branch '22952' of github.com:dannytsang/home-assistant.io into 22952

ndudde67 commented on source/getting-started/concepts-terminology.markdown:
```diff
@@ -0,0 +1,37 @@
+
+![Home Assistant Device](/images/getting-started/home-assistant-device.png)
+
+### Automations
```
Could this be expanded to mention Scenes?

ndudde67 reviewed (dismissed):
Other than maybe adding "Scenes" to the Automations section of Terminology, I think this is an excellent addition to the info that greets newcomers to the platform!

dannytsang committed:
Added scripts and scenes.

Replaced steps with action which is the same word used in other pages like /integrations/script/

dannytsang committed:
Merge branch 'home-assistant:current' into 22952

dannytsang commented:
Hi @frenck

Any thoughts on the PR?

frenck commented:
I am out for holiday with my family. Will take a bit for me to get back at this.

frenck commented:
not stale dear bot

c0ffeeca7 commented:
Hi @dannytsang,
:heart: Love this.
Do you think you could split this PR into 2 parts? To have the terminology section in a separate PR?
I could look at the terminology right away but I might need some support for the integration section. Splitting the PR could help getting this merged more quickly.

c0ffeeca7 reviewed (changes requested):
Hi @dannytsang,
:heart: Love this.
Do you think you could split this PR into 2 parts? To have the terminology section in a separate PR?
I could look at the terminology right away but I might need some support for the integration section. Splitting the PR could help getting this merged more quickly.

dannytsang commented:
@c0ffeeca7 I have created #26529 for the concepts and terminology page. In terms of this PR for the integrations, how do I move forward with this?

I think I need to remove changes that are part of #26529 so there won't be any conflicts when merging. Is there anything else or better way of doing this?

frenck commented:
> I think I need to remove changes that are part of https://github.com/home-assistant/home-assistant.io/pull/26529 so there won't be any conflicts when merging. Is there anything else or better way of doing this?

There is no better way, removing those changes from this PR is the way to go.

../Frenck

dannytsang committed:
Removed concepts and terminology page because it has been submitted as a separate PR.

Updated navigation to match.

dannytsang committed:
Missed removing concepts and terminology in getting started index.

dannytsang committed:
Merge branch 'current' into 22952

dannytsang committed:
Post merge changes:

* Moved integration details from onboarding to it's own page.
* Added integration page in getting started after concepts & terminology.
* Added tooltips to integration page.
* Change links to devices & services to Microsoft style.

c0ffeeca7 commented on source/getting-started/integration.markdown:
```diff
@@ -0,0 +1,31 @@
+
+![Google Cast discovered confirmation](/images/getting-started/google-cast-discovered.png)
+
+It now asks for the network IP address of the device. This is beyond the scope of this article because each network is different however the router is usually a good place to look. In this example, it's *192.168.1.99*.
```
```suggestion
It now asks for the network IP address of the device. Describing how to identify a devices IP address is beyond the scope of this article because each network is different. However, the router is usually a good place to look. In this example, it's *192.168.1.99*.
```

c0ffeeca7 commented on source/getting-started/integration.markdown:
```diff
@@ -0,0 +1,31 @@
+
+When you're done, navigate back to the web interface and voila, your devices are ready for you to control. You can always go back to add and remove integrations in the future.
+
+If you get stuck, need more information or want to see what the setup process is like, use the [integration documentation](/integrations).
```
```suggestion
If you get stuck, need more information, or want to see what the setup process is like, use the [integration documentation](/integrations).
```

c0ffeeca7 reviewed (changes requested):
nice guide!
One small tweak.

dannytsang committed:
Additional changes from feedback.

c0ffeeca7 commented:
Thank you, @dannytsang! :+1:
